PowerShellGet/PSModule.psm1

```diff
@@ -2873,18 +3005,65 @@ function Publish-Script
                                -CallerPSCmdlet $PSCmdlet `
                                -ErrorCategory InvalidOperation
                 }
-                elseif(-not $Force -and ($currentPSGetItemInfo.Version -gt $PSScriptInfo.Version))
+
+                $galleryScriptVer,$galleryScriptPrerelease = $galleryScriptVersion -split '-',2
+                $toPublishScriptVer,$toPublishScriptPrerelease = $toPublishScriptVersion -split '-',2
+
+                if ($galleryScriptVer -eq $toPublishScriptVer)
+                {
+                    # Prerelease strings will not both be null, otherwise would have terminated already above
+
+                    if (-not $Force -and (-not $galleryScriptPrerelease -and $toPublishScriptPrerelease))
+                    {
+                        # User is trying to publish a new Prerelease version AFTER publishing the stable version.
+                        $message = $LocalizedData.ScriptVersionShouldBeGreaterThanGalleryVersion -f ($scriptName,
+                                                                                                     $PSScriptInfo.Version,
+                                                                                                     $currentPSGetItemInfo.Version,
+                                                                                                     $currentPSGetItemInfo.RepositorySourceLocation)
+                        ThrowError -ExceptionName "System.InvalidOperationException" `
+                                -ExceptionMessage $message `
+                                -ErrorId "ScriptVersionShouldBeGreaterThanGalleryVersion" `
+                                -CallerPSCmdlet $PSCmdlet `
+                                -ErrorCategory InvalidOperation

```

Comment:
Sure, I'll add that.

---
In reply to: [132598316](https://github.com/rebro-msft/PowerShellGet/pull/1#discussion_r132598316) [](ancestors = 132598316)